src/main/views/crown-daily-list.njk

```diff
@@ -0,0 +1,210 @@
+          {% if courtRoom['exclude'] !== true %}
+            {% set hearingCount = 1 %}
+            {% for session in courtRoom['session'] %}
+            {% set sessionCount = sessionCount + 1 %}
+
+            {% set sessionChannel = '' %}
+            {% for channel in session['sessionChannel'] %}
+              {% set sessionChannel = sessionChannel + ', ' + channel %}
+            {% endfor %}
+
+            {% set sessionChannel = sessionChannel.slice(1) %}
+
+            <div class="govuk-accordion__section govuk-accordion__section--expanded">
+              <div class="govuk-accordion__section-header">
+                <h2 class="govuk-accordion__section-heading">
+                <span class="govuk-accordion__section-button" id="accordion-default-heading-{{ sessionCount }}">
+                    {% if session['formattedJudiciaries'] | length and courtList['unallocatedCases'] !== true %}
+                      {{ courtRoom['courtRoomName'] }}: {{ session['formattedJudiciaries'] }}
+                    {% else %}
+                      {{ courtRoom['courtRoomName'] }}
+                    {% endif %}
+                </span>
+                </h2>
+              </div>
+              <div id="accordion-default-content-{{ sessionCount }}" class="govuk-accordion__section-content" aria-labelledby="accordion-default-heading-{{ sessionCount }}">
+                <div class="parent-box overflow-table">
+                  <table class="govuk-table">
+                    <thead class="govuk-table__head">
+                    <tr class="govuk-table__row">
+                      <th scope="col" class="govuk-table__header no-wrap">{{ sittingAt }}</th>
+                      <th scope="col" class="govuk-table__header no-wrap">{{ caseReference }}</th>
+                      <th scope="col" class="govuk-table__header no-wrap">{{ defendantName }}</th>
+                      <th scope="col" class="govuk-table__header no-wrap">{{ hearingType }}</th>
+                      <th scope="col" class="govuk-table__header no-wrap">{{ duration }}</th>
+                      <th scope="col" class="govuk-table__header no-wrap">{{ prosecutingAuthority }}</th>
+                    </tr>
+                    </thead>
+
+                    <tbody class="govuk-table__body">
+                    {% for sitting in session['sittings'] %}
+                      {% set hearingPlatform = '' %}
+                      {% if sitting['channel'].length === 0 %}
+                        {% set hearingPlatform = sessionChannel %}
+                      {% else %}
+                        {% for channel in sitting['channel'] %}
+                          {% set hearingPlatform = hearingPlatform + ', ' + channel %}
+                        {% endfor %}
+                        {% set hearingPlatform = hearingPlatform.slice(1) %}
+                      {% endif %}
+                      {% set durationAsHours = 0 %}
+                      {% if sitting['durationAsHours'] !== 0 %}
```

Comment:
This duration section feels better to be done as javascript if possible I think we may have a duration function somewhere in the data manipulation service

Reply:
refactor the code and move the code to javascript helper class.